### ISSUE TITLE
camelcase false positive with `global` and `new`

### ISSUE THREAD

**Issue by silverwind** [eslint/eslint#7363]
**Tell us about your environment**

* **ESLint Version:** v3.7.1
* **Node Version:** v6.7.0
* **npm Version:** 3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```
rules:
  camelcase: 2
```

**What did you do? Please include the actual source code causing the issue.**
```js
/* global My_Global */
new My_Global;
```
or
```js
/* global My_Global */
new Object(My_Global);
```

**What did you expect to happen?**
No error should be given, based on the fact that in a browser, on can usually not change the name of globals provided. Interestingly, the bug only happens when `new` is involved, e.g. the code passes once `new` is removed.

**What actually happened? Please include the actual, raw output from ESLint.**
```
  2:5  error  Identifier 'My_Global' is not in camel case  camelcase
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. I can reproduce this.

I'm not sure what the intended behavior is here, but it seems like we should be consistent between `My_Global()` and `new My_Global()`.

**Comment by Kovensky** [comment]
It also flags things like `__webpack_public_path__`, which are meta-globals (the webpack compiler replaces it with something else).
